### PR TITLE
INVGOV-13228 /report/{content_id} Node.js API endpoint

### PR DESCRIPTION
**Pull request by klauselhai**
https://io1dev.atlassian.net/browse/INVGOV-13228

### CODE REVIEW

File: api.swagger.yaml [hunk 1]
```diff
@@ -5611,6 +5611,82 @@ paths:
...
+        '422':
+          description: 'The content/comment was not reported due to Drupal validation failure or JSON schema validation failure.'
+      security:
+        - SessionCookieHeader: []
+        - APIKeyHeader: []
+        - CognitoJwt: [ invotra/report:write ]
+    delete:
+      tags:
+        - report
+      description: 'Unreport a piece of content/comment.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5644]: Unreport a piece of content/comment.

->

Remove report flag from a piece of content/comment.

File: api.swagger.yaml [hunk 2]
```diff
@@ -5611,6 +5611,82 @@ paths:
...
+      responses:
+        '200':
+          description: 'Report removed successfully.'
+        '400':
+          description: 'Could not unreport this content/comment due to malformed JSON.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5649]: Could not unreport this content/comment due to malformed JSON.

->

Could not remove report flag due to malformed JSON.

File: api.swagger.yaml [hunk 3]
```diff
@@ -5611,6 +5611,82 @@ paths:
...
+        '403':
+          description: 'User is not authorized to unreport this content/comment.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5651]: User is not authorized to remove report flag.

File: api.swagger.yaml [hunk 4]
```diff
@@ -5611,6 +5611,82 @@ paths:
...
+        '422':
+          description: 'The report could not be removed due to Drupal validation failure or JSON schema validation failure.'
+      security:
+        - SessionCookieHeader: []
+        - APIKeyHeader: []
+        - CognitoJwt: [ invotra/report:write ]
+    get:
+      description: 'Check if a piece of content/comment is reported or not.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5659]: Check if a piece of content/comment is reported or not.

->

Check if a piece of content/comment has been reported or not.

File: api.swagger.yaml [hunk 5]
```diff
@@ -19133,3 +19209,9 @@ components:
         current_user_saved:
           type: boolean
           default: false
+    content-report-get-response-schema:
+      type: object
+      properties:
+        current_user_reported:
+          type: boolean
+          default: false
```
Comment (ParryLuke) [inline api.swagger.yaml:19217]: Newline needed at end of file

**Review by ParryLuke — CHANGES_REQUESTED** [review]

**Review by ParryLuke — APPROVED** [review]

**Comment by MireilleHelle** [comment]
Closing task as we’re now focusing on building only API gaps for which Invotra Consulting have a direct need.

**Pull request closed without merging**